lib/iris/experimental/um.py

```diff
@@ -590,12 +590,16 @@ def _update_fixed_length_header(self):
         # Now deal with the LOOKUP and DATA components.
         if self.fields:
             header.lookup_start = word_number
-            lengths = set(field.num_values() for field in self.fields)
-            if len(lengths) != 1:
-                msg = 'Inconsistent header lengths - {}'.format(lengths)
+            header_lengths = [field.num_values() for field in self.fields]
+            if len(set(header_lengths)) != 1:
+                msg = 'Inconsistent header lengths - {}'.format(header_lengths)
```

Comment:
- There's no need to switch `lengths` from being a `set` to being a `list`. (NB. A set supports `pop()`.)
- Given the ambiguity of the word "header" in the context of a FieldsFile variant, adding a `header_` prefix is *less* clear than just `lengths`. Perhaps `unique_num_values`? And then `num_values = unique_num_values.pop()`?